File: 2404.cpp
// Crucigrama
// https://jv.umsa.bo/problem.php?id=2404

#include <iostream>
#include <vector>
#include <string>

using namespace std;

string searchWord(const vector<string>& grid, const string& word) {
    int rows = grid.size();
    int cols = grid[0].length();

    // Buscar horizontalmente
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols - word.length() + 1; j++) {
            if (grid[i].substr(j, word.length()) == word) {
                return word + " (" + to_string(i) + ", " + to_string(j) + ") -H";
            }
        }
    }

    // Buscar verticalmente
    for (int i = 0; i < rows - word.length() + 1; i++) {
        for (int j = 0; j < cols; j++) {
            string column;
            for (int x = i; x < i + word.length(); x++) {
                column += grid[x][j];
            }
            if (column == word) {
                return word + " (" + to_string(i) + ", " + to_string(j) + ") -V";
            }
        }
    }

    // Buscar en diagonal de izquierda a derecha y de arriba hacia abajo
    for (int i = 0; i < rows - word.length() + 1; i++) {
        for (int j = 0; j < cols - word.length() + 1; j++) {
            string diagonal;
            for (int x = 0; x < word.length(); x++) {
                diagonal += grid[i + x][j + x];
            }
            if (diagonal == word) {
                return word + " (" + to_string(i) + ", " + to_string(j) + ") -D";
            }
        }
    }

    return "";
}

int main() {
    vector<string> grid;
    string line;

    getline(cin, line);
    grid.push_back(line);
    for (int i = 0; i < line.length() - 1; i++) {
        getline(cin, line);
        if (line.empty()) {
            break;
        }
        grid.push_back(line);
    }

    string word;

    while (cin >> word) {
        if (word.empty()) {
            break;
        }
        string result = searchWord(grid, word);
        cout << result << endl;
    }

    return 0;
}

File: jvPY/2404.py
# Crucigrama
# https://jv.umsa.bo/problem.php?id=2404

def search_word(grid, word):
    rows = len(grid)
    cols = len(grid[0])

    # Buscar horizontalmente
    for i in range(rows):
        for j in range(cols - len(word) + 1):
            if grid[i][j:j+len(word)] == word:
                return f"{word} ({i}, {j}) -H"

    # Buscar verticalmente
    for i in range(rows - len(word) + 1):
        for j in range(cols):
            column = [grid[x][j] for x in range(i, i+len(word))]
            if column == list(word):
                return f"{word} ({i}, {j}) -V"

    # Buscar en diagonal de izquierda a derecha y de arriba hacia abajo
    for i in range(rows - len(word) + 1):
        for j in range(cols - len(word) + 1):
            diagonal = [grid[i+x][j+x] for x in range(len(word))]
            if diagonal == list(word):
                return f"{word} ({i}, {j}) -D"

    return None


# Lectura del crucigrama
grid = []
cad = input()
grid.append(cad)
for i in range(len(cad)-1):
    line = input().strip()
    if line == "":
        break
    grid.append(line)

# Búsqueda de las palabras
while True:
    try:
        word = input().strip()
        if word == "":
            break
        result = search_word(grid, word)
        print(result)
    except EOFError:
        break
